src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs

```diff
@@ -332,31 +333,45 @@
 (react/defc- WorkflowDetails
   {:render
    (fn [{:keys [state props]}]
-     (let [server-response (:server-response @state)]
+     (let [metadata-response (:metadata-response @state)
+           cost-response (:cost-response @state)]
        (cond
-         (nil? server-response)
+         (nil? metadata-response)
          (spinner "Loading workflow details...")
-         (not (:success? server-response))
-         (style/create-server-error-message (:response server-response))
+         (not (:success? metadata-response))
+         (style/create-server-error-message (:response metadata-response))
+         (nil? cost-response)
```

Comment:
Only checks for cost errors after doing so the main metadata errors.  This seems like the right priority to me.